snowfakery/data_generator_runtime.py

```diff
@@ -356,22 +346,38 @@ def __init__(
             if not name.startswith("_") and name != "context"
         }
 
-        self.templates = templates
+        self.statements = statements
+
+    def faker_template_library(self, locale):
+        rc = self.faker_template_libraries.get(locale)
+        if not rc:
+            rc = FakerTemplateLibrary(self.faker_providers, locale)
+            self.faker_template_libraries[locale] = rc
+        return rc
 
     def loop_over_templates_until_finished(self, continuing):
         finished = False
         runtimecontext = RuntimeContext(interpreter=self)
         while not finished:
-            self.loop_over_templates_once(runtimecontext, continuing)
-            finished = runtimecontext.check_if_finished()
+            self.loop_over_templates_once(self.statements, continuing)
+            finished = self.current_context.check_if_finished()
             continuing = True
             self.globals.reset_slots()
 
-    def loop_over_templates_once(self, runtimecontext, continuing):
-        for template in self.templates:
-            should_skip = template.just_once and continuing
-            if not should_skip:
-                template.generate_rows(self.output_stream, runtimecontext)
+    def loop_over_templates_once(self, statement_list, continuing: bool):
+        for statement in statement_list:
+            if statement.__class__.__name__ == "ObjectTemplate":
+                template = statement
+                should_skip = template.just_once and continuing
+                if not should_skip:
+                    template.generate_rows(self.output_stream, self.current_context)
+            elif statement.__class__.__name__ == "VariableDefinition":
+                vardef = statement
+                self.current_context.variable_definitions()[
+                    vardef.varname
+                ] = vardef.evaluate(self.current_context)
+            else:
+                assert 0, statement  # pragma: no cover
```

Comment:
The if/else checking class types makes me think this should be implemented as an `evaluate` method of the statement object (passed the interpreter as an arg, I suppose).